R/report.R

```diff
@@ -0,0 +1,93 @@
+#' Data report for datasets with 'mdate' variables
+#'
+#' Create a properly formatted data report for datasets which contain 'mdate'
+#' class objects, alongside other object classes.
+#' @param data A `{tibble}` or a `{data.frame}`.
+#' @return A data report of class 'mreport'.
+#' @importFrom dplyr %>%
+#' @importFrom scales percent_format
+#' @importFrom stats na.omit
+#' @details 'mreport' displays the variable's name,
+#' the variable type, the number of missing observations for variable,
+#' the percentage of missing observations in variable,
+#' the minimal value in variable, and the maximal value in variable.
+#' @examples
+#' mreport(battles)
+#' @export
+mreport <- function(data) UseMethod("mreport")
+
+#' @export
+mreport.default <- function(data) {
+  if (!is.data.frame(data)) {
+    stop("Data must be a `data.frame` or `tibble`.")
+  }
+  rows     <- nrow(data)
+  cols     <- ncol(data)
+  varnames <- names(data)
+  datatype <- unlist(lapply(data, class))
+  counts   <- unlist(lapply(data, length))
+  mvalues    <- unlist(lapply(data, function(z) sum(is.na(z))))
+  mvaluesper <- round((mvalues / counts) * 100, 2)
+  minv <- suppressWarnings(unlist(lapply(data, function(x) {
```

Comment:
At the moment this does not work for treaty text variables which are lists, and in any case it doesn't make much sense to calculate a min and max for these. I also think the min and max might not be appropriate for understanding other categorical variables like `AgreementType` and `DocType` in `manytrade` and `manyenviron`, maybe for these it might be better to have an indication of the number of types of observations in these variables? Eg. for `DocType` we have bilateral, plurilateral, and multilateral, so the table could show that there are 3 categories of observations?

Reply:
Thank you so much for the feedback @jaeltan , I appreciate! I think you are right about min and max, I removed these for now. I will look into listing the categories of factor variables, but I imagine the output may be too big for a column in the report and make it look "uglier"... Now `mreport()`should also work well for text databases (try lapply(manytrade::texts, mreport) for example). Please let me know if you have more suggestions!